srcpkgs/dkms/files/kernel.d/dkms.prerm

```diff
@@ -1,21 +1,19 @@
-#!/bin/bash
+#!/bin/sh
 
-# We're passed the version of the kernel being installed
+# We're passed the version of the kernel being removed
 PKGNAME="$1"
 VERSION="$2"
 
 if [ -x /usr/bin/dkms ]; then
-while read line; do
-   name=`echo "$line" | awk '{print $1}' | sed 's/,$//'`
-   vers=`echo "$line" | awk '{print $2}' | sed 's/,$//'`
-   arch=`echo "$line" | awk '{print $4}' | sed 's/:$//'`
-   echo "dkms: removing: $name $vers (${PKGNAME}-${VERSION}) ($arch)" >&2
-   /usr/bin/dkms remove -q -m $name -v $vers -k ${VERSION} -a $arch
-done < <(/usr/bin/dkms status -k ${VERSION} 2>/dev/null | grep ": installed")
+    /usr/bin/dkms status -k "$VERSION" 2>/dev/null | while IFS=,:/ read -r name vers _ arch status; do
```

Comment:
`IFS` needs to accommodate whitespace, too. See comment in the `xbps-trigger` script.